### ISSUE TITLE
pollInterval should be a Duration (semantically)

### ISSUE THREAD

**Issue by mazei513** [zino-app/graphql-flutter#558]
`QueryOptions` describes `pollInterval` as:

```dart
  /// The time interval (in milliseconds) on which this query should be
  /// re-fetched from the server.
  int pollInterval;
```

However, looking into `QueryScheduler`, it is used as seconds.

```dart
  void startPollingQuery(
    WatchQueryOptions options,
    String queryId,
  ) {
    assert(options.pollInterval != null && options.pollInterval > 0);

    registeredQueries[queryId] = options;

    final Duration interval = Duration(
      seconds: options.pollInterval,
    );

    if (intervalQueries.containsKey(interval)) {
      intervalQueries[interval].add(queryId);
    } else {
      intervalQueries[interval] = <String>[queryId];

      _pollingTimers[interval] = Timer.periodic(
        interval,
        (Timer timer) => fetchQueriesOnInterval(timer, interval),
      );
    }
  }
```

On package version 3.0.0.

**Comment by mazei513** [comment]
Would like to suggest changing `pollInterval` to be a `Duration` instead.

**Comment by cihadturhan** [comment]
Hello,
I agree!
It says pollinterval is in `milliseconds` here:
https://github.com/zino-app/graphql-flutter/blob/master/packages/graphql/lib/src/core/query_options.dart#L129

but it's in `seconds`:
https://github.com/zino-app/graphql-flutter/blob/master/packages/graphql/lib/src/scheduler/scheduler.dart#L44

I tried 1000 and waited forever :)

I'm quite busy right now, but I'd like to send a PR if I have time in a few days.

**Comment by micimize** [comment]
closed in `4.0.0-beta.1` #648
